repository/utils/utils.go

```diff
@@ -0,0 +1,59 @@
+package utils
+
+import (
+	"errors"
+	"fmt"
+
+	reperr "github.com/jan-zabloudil/release-manager/repository/errors"
+	postgrestgo "github.com/nedpals/postgrest-go/pkg"
+	"github.com/nedpals/supabase-go"
+)
+
+func ValidateSingleRecordFetch[T any](s []T) error {
+	if len(s) != 1 {
+		return reperr.ErrMultipleOrNoRecordsReturnedAfterWriteOperation
+	}
+
+	return nil
+}
+
+func WrapSupabaseDBErr(err error) error {
+	var postgreErr *postgrestgo.RequestError
+
+	if errors.As(err, &postgreErr) {
+		switch postgreErr.HTTPStatusCode {
+		case 406:
+			return fmt.Errorf("%w: %s", reperr.ErrResourceNotFound, err.Error())
+		}
+	}
+
+	return fmt.Errorf("%w: %s", reperr.GeneralErr, err.Error())
+}
+
+func WrapSupabaseAuthErr(err error) error {
+	var supabaseErr *supabase.ErrorResponse
+	if errors.As(err, &supabaseErr) {
+		switch supabaseErr.Code {
+		case 401:
+			return fmt.Errorf("%w: %s", reperr.ErrUserAuthenticationFailed, err.Error())
+		case 404:
+			return fmt.Errorf("%w: %s", reperr.ErrUserAuthenticationFailed, err.Error())
+		}
+	}
+
+	return fmt.Errorf("%w: %s", reperr.GeneralErr, err.Error())
+}
+
+func WrapSupabaseAdminErr(err error) error {
```

Comment:
I don't understand the name of this function. Why is this `AdminErr`? I get DBErr and AuthErr from the previous functions, but here it's not immediately apparent to me.

Reply:
Auth endpoint is called in order to authenticate user. User’s bearer token is provided in the authorization header.
**Error responses are following:** 

- 401 = invalid bearer token -> ReleaseManager returns 401
- 404 = user not found (valid token but user no longer exists in the database) -> ReleaseManager returns 401.

---
Admin endpoins are called in order to retrive users. Supabase secret key must be provided to access auth data.
**Error responses are following:**
- 401 = cannot access resource (invalid supabase secret key provided or only supabase anon key provided) -> ReleaseManager returns 500 (cannot access db because bad credentials were provided in the config)
- 404 = user not found (no user found for provided uuid) -> ReleaseManager returns 404

I needed to distinquish between those cases to return correct response. 
And I find better to create separate function for Auth and Admin than checking error text as well. I find it better to rely only on returned status code.